config-model/src/main/java/com/yahoo/vespa/model/application/validation/change/search/AttributeChangeValidator.java

```diff
@@ -136,27 +136,37 @@ private static void validateAttributePredicate(ClusterSpec.Id id,
         }
     }
 
-    private static <T> void validateAttributeProperty(ClusterSpec.Id id,
+    private <T> void validateAttributeProperty(ClusterSpec.Id id,
                                                       Attribute current, Attribute next,
                                                       Function<Attribute, T> settingValueProvider, String setting,
                                                       List<VespaConfigChangeAction> result) {
         T currentValue = settingValueProvider.apply(current);
         T nextValue = settingValueProvider.apply(next);
         if ( ! Objects.equals(currentValue, nextValue)) {
             String message = String.format("change property '%s' from '%s' to '%s'", setting, currentValue, nextValue);
+            if (hasHnswIndex(current) && hasHnswIndex(next))
```

Comment:
Is this intended to catch _adding_ a hnsw index? If so, should the first hasHnswIndex be negated?

Reply:
It catches only changes to `distance-metric` when already having a hnsw index. I wasn't intending to require a validation override when adding a hnsw index in this PR, sounds a bit restrictive, but I am not sure...

Reply:
Agreed, was just me not understanding the code here then :)